lib/miq_tools_services/github/message_builder.rb

```diff
@@ -0,0 +1,56 @@
+require 'stringio'
+
+module MiqToolsServices
+  class Github
+    class MessageBuilder
+      attr_accessor :header, :continuation_header
+
+      GITHUB_COMMENT_BODY_MAX_SIZE = 65_535
+
+      def initialize(header = nil, continuation_header = nil)
+        @header = header
+        @continuation_header = continuation_header
+        @lines = []
+      end
+
+      def write(line)
+        if line.length >= GITHUB_COMMENT_BODY_MAX_SIZE
+          raise ArgumentError, "line length must be less than #{GITHUB_COMMENT_BODY_MAX_SIZE}"
```

Comment:
For the 0% of the times this will occur, it's not worth the coding effort of finding a line break or doing it cleanly.